crates/iroha_data_model/src/block.rs

```diff
@@ -208,88 +228,108 @@ impl SignedBlock {
         block.hash()
     }
 
+    /// Add additional signature to this block
+    #[cfg(feature = "transparent_api")]
+    pub fn sign(&mut self, private_key: &iroha_crypto::PrivateKey, signatory: usize) {
+        let SignedBlock::V1(block) = self;
+
+        block.signatures.push(BlockSignature(
+            signatory as u64,
+            SignatureOf::new(private_key, &block.payload.header),
+        ));
+    }
+
     /// Add signature to the block
     ///
     /// # Errors
     ///
     /// if signature is invalid
     #[cfg(feature = "transparent_api")]
-    pub fn add_signature(
-        &mut self,
-        signature: BlockSignature,
-        public_key: &iroha_crypto::PublicKey,
```

Comment:
validity of signatures is not an invariant of `SignedBlock` so there is no need to verify this here